lib/components/viewers/route-viewer.js

```diff
@@ -51,55 +79,121 @@ class RouteViewer extends Component {
+    const routesArray = routes ? Object.values(routes) : []
+
+    const filteredAndSortedRoutes = routesArray
+      .filter(
+        (route) =>
+          // If the filter isn't defined, don't check.
+          (!agency || agency === route.agencyName) &&
+          (!mode || mode === route.mode) &&
+          // If user search is active, filter here
+          (!search ||
+            ((route.longName && route.longName.includes(search)) ||
+              (route.shortName && route.shortName.includes(search))))
+      )
+      .sort(coreUtils.route.makeRouteComparator(transitOperators))
+
+    // get agency and mode lists from route list
+    const agencies = Array.from(
+      new Set(routesArray.map((route) => route.agencyName || route.agency.name))
+    )
+      .filter((agency) => agency !== undefined)
+      .sort()
+
+    const modes = Array.from(
+      new Set(
+        routesArray
+          .filter((route) => route.mode !== undefined)
+          .map((route) => route.mode.toLowerCase())
       )
-      : []
+    )
+      // Correct capitalization
+      .map((mode) => mode[0].toUpperCase() + mode.substr(1).toLowerCase())
+      .sort()
+
     return (
       <div className='route-viewer'>
         {/* Header Block */}
         <div className='route-viewer-header'>
           {/* Back button */}
           {!hideBackButton && (
             <div className='back-button-container'>
-              <Button
-                bsSize='small'
-                onClick={this._backClicked}
-              ><Icon type='arrow-left' />Back</Button>
+              <Button bsSize='small' onClick={this._backClicked}>
+                <Icon type='arrow-left' />
+                Back
+              </Button>
             </div>
           )}
 
           {/* Header Text */}
           <div className='header-text'>
             {languageConfig.routeViewer || 'Route Viewer'}
           </div>
-          <div className=''>
-            {languageConfig.routeViewerDetails}
-          </div>
-          <div style={{ clear: 'both' }} />
+          <div className=''>{languageConfig.routeViewerDetails}</div>
+          <section className='search-and-filter'>
+            <span className='routeFilter'>
+              <Icon type='filter' />
+              <select
+                id='agency'
+                onBlur={this.onFilterChange}
+                onChange={this.onFilterChange}
+              >
+                <option value=''>All agencies</option>
+                {agencies.map((agency) => (
+                  <option key={agency}>{agency}</option>
+                ))}
+              </select>
+              <select
+                id='mode'
+                onBlur={this.onFilterChange}
+                onChange={this.onFilterChange}
+              >
+                <option value=''>All modes</option>
+                {modes.map((mode) => (
+                  <option key={mode} value={mode.toUpperCase()}>
+                    {mode}
```

Comment:
Let's go ahead and add react-intl (everywhere in this file) if you can coordinate/get the OK from @binh-dam-ibigroup.

Reply:
@miles-grant-ibigroup For the OTP mode of each route (e.g. `BUS`, `SUBWAY`, etc), you can access the localized strings using the message id `common.otpTransitModes.${mode.toLowerCase}`.

Reply:
Internationalized in ae4bf350860169e3e74de8c5596047fdb3add159